src/context/BasketContext.tsx

```diff
@@ -21,7 +23,37 @@ export function BasketContextProvider(props: { children: JSX.Element }): JSX.Ele
 
     const addToBasket = (bi: BasketItem) => {
         const next = basket.clone();
-        next.addItem(bi);
+        const productExistence = next.items.find(
+            (item) => item.product.upc === bi.product.upc
+          );
+          if (productExistence) {
+            productExistence.quantity = productExistence.quantity + 1;
```

Comment:
Shouldnt this summarize both items quantities instead of constant 1?

For instance, having:

```
BasketItem {
  upc: 1,
  quantity: 2
}
```

```
BasketItem {
  upc: 1,
  quantity: 1
}
```

We should have quantity as `3` no?

Reply:
But my button will always add 1, in this way I am sure that I will always add by one